src/recommendationservice/recommendation_server.py

```diff
@@ -35,6 +36,35 @@
 from logger import getJSONLogger
 logger = getJSONLogger('recommendationservice-server')
 
+def initStackdriverProfiling():
+  enable_profiler = None
+  project_id = None
+  try:
+    enable_profiler = os.environ["ENABLE_PROFILER"]
+    project_id = os.environ["GCP_PROJECT_ID"]
+  except KeyError:
+    # Environment variable not set
+    pass
+  if enable_profiler != "1":
+    logger.info("Skipping Stackdriver Profiler Python agent initialization. Set environment variable ENABLE_PROFILER=1 to enable.")
+    return
+
+  for retry in xrange(1,4):
+    try:
+      if project_id:
```

Comment:
Users will never set `GCP_PROJECT_ID` env var in this demo even when they're running on GCP.

It's Stackdriver SDK's responsibility to pick up the GCP project ID from the metadata server if the value is not specified. If this is not the case, we should block the PR on updating the library first.

Reply:
It is not required. If it is provided, it will be used. If its not, start() will try to get it from metadata engine. 
Users can run profiler when not in GCP environments, and they will need to provide the project id in that scenario since there is not metadata server available. 
When in GCP, its is not required.